### ISSUE TITLE
XML output does not include cross-references in code-fragments

### ISSUE THREAD

**Issue by lukaspj** [doxygen/doxygen#8357]
**Describe the bug**
I'm running Doxygen on the Torque3D code-base, and generating an XML file and HTML site.

The HTML site has cross-references in code-fragments:
![image](https://user-images.githubusercontent.com/2369614/105848244-01f34d00-5fdf-11eb-9cd6-784890a54ddc.png)

But XML does not (excerpt):
```xml
<para><programlisting><codeline><highlight class="normal">class<sp/>mySubClass<sp/>:<sp/>public<sp/>SimObject<sp/>{</highlight></codeline>
<codeline><highlight class="normal"><sp/><sp/><sp/><sp/>typedef<sp/>SimObject<sp/>Parent;</highlight></codeline>
<codeline><highlight class="normal"><sp/><sp/><sp/><sp/>...</highlight></codeline>
</programlisting></para>
```

**Expected behavior**
I expected <ref> elements inside the <programlisting> corresponding to the cross-references found in HTML

**To Reproduce**
MWE with example output included:
[doxygen-fragments.zip](https://github.com/doxygen/doxygen/files/5876002/doxygen-fragments.zip)

**Version**
Doxygen: 1.8.17 (fetched via apt-get) and 1.9.1 (downloaded from website)
Running inside a Docker container with Ubuntu 20.04 as base

**Additional context**
I don't mind poking around in Doxygen myself and see if I can solve it myself, but I'm not familiar with the code-base and could really use some guidance. Also my time is limited, so I might not be able to solve it.

**Comment by albert-github** [comment]
As noted with #8358:
- I didn't see any html files as `GENERATE_HTML=NO` so how are they generated?
- the xml snippet, from which file is this?
- the html snippets, from which files are these?

**Comment by lukaspj** [comment]
I updated this issue with a better MWE as well.

**Comment by lukaspj** [comment]
Tested it in 1.9.1 and it is still an issue

**Comment by lukaspj** [comment]
Added a MWE as a zip file

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8359

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by lukaspj** [comment]
Sounds great, I look forward to the official release!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
